### PR TITLE
1598 by Inlead: Implement ding_dams into DDB CMS.

### PR DESCRIPTION
**Pull request by inlead**
http://platform.dandigbib.org/issues/1598

### CODE REVIEW

**Comment by guddo** [comment]
This PR is made for the 7.x-2.5.1 release.

File: themes/ddbasic/scripts/ddbasic.common.js [hunk 1]
```diff
@@ -84,6 +84,17 @@
...
+    $('.primary-content video, .primary-content audio').on('loadedmetadata', function (i) {
+      // If video size bigger then window size, it should be resized.
+      var video = i.currentTarget;
+      var width = $(this).parent().width() * 0.8;
+
+
+      if (video.videoWidth > width) {
+        $(this).width(width * 0.9);
+      }
+    });
```
Comment (kasperg) [inline themes/ddbasic/scripts/ddbasic.common.js]: Why is this added to `ddbasic.common.js` instead of in `ding_dams`?

Also: Again 0.8 and 0.9 is used. Why is this?

File: modules/ding_dams/css/ding_dams.media_browser.css [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    background: #eee;
+    border: 1px solid #CCCCCC;
+    box-shadow: inset 0 0 15px rgba(0,0,0,.1), inset 0 0 0 1px rgba(0,0,0,.05);
+    display: inline-block;
+    padding: 5px;
+    position: relative;
+    min-width: 100px;
+    max-width: 100px;
+    min-height: 100px;
+    max-height: 100px;
```
Comment (kasperg) [inline modules/ding_dams/css/ding_dams.media_browser.css]: `min-width` and `max-width` seems to be repeated here.

File: modules/ding_dams/css/ding_dams.media_edit.css [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+div#modal-content {
+  display:block;
+  /* @TODO: Disallow !important - Use of !important. */
+  width: 700px !important;
+  /* @TODO: Disallow !important - Use of !important. */
+  height: 330px !important;
+  /* @TODO: Disallow !important - Use of !important. */
+  margin: 0 !important;
+  /* @TODO: Disallow !important - Use of !important. */
+  padding: 0 5px !important;
```
Comment (kasperg) [inline modules/ding_dams/css/ding_dams.media_edit.css]: There are a lot of `!important` here. It should be avoided as mentioned in `@TODO`.

If you for some reason think it is necessary then please add a comment explaining why.

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 65c6616d2967762b59ca7a6134a5c01e518008c9
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 65c6616d2967762b59ca7a6134a5c01e518008c9
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 65c6616d2967762b59ca7a6134a5c01e518008c9
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: ea83733ab52653d8c173bc0a02cfab32cf0f8919
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 65c6616d2967762b59ca7a6134a5c01e518008c9
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: ea83733ab52653d8c173bc0a02cfab32cf0f8919
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 65c6616d2967762b59ca7a6134a5c01e518008c9
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: aaa0c578d6494ef4d3c9c825f90cbcfcfa885dd7
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 65c6616d2967762b59ca7a6134a5c01e518008c9
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: aaa0c578d6494ef4d3c9c825f90cbcfcfa885dd7
  * The first line of your commit message (the subject) must follow this format: [issue number]: [short summary of change]

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

File: modules/ding_dams/wysiwyg_plugins/js/wysiwyg-dams-video.js [hunk 4]
```diff
@@ -131,6 +131,8 @@
           var image = document.createElement('img');
           image.src = Drupal.settings.ding_dams.icon_path + '/doc_flv.png';
           a.appendChild(image);
+
+          markup = a.outerHTML;
```
Comment (avoykov) [inline modules/ding_dams/wysiwyg_plugins/js/wysiwyg-dams-video.js]: Duplicate

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: b8d2d9865d052a5474699748bc4a9681ccee7a8e
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: b8d2d9865d052a5474699748bc4a9681ccee7a8e
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 3313ce2668dcb1872c18007f327a7f7dac5c6ef9
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 3313ce2668dcb1872c18007f327a7f7dac5c6ef9
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 4b0542b59ce479365af81c94747b2de4b3b276f3
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 4b0542b59ce479365af81c94747b2de4b3b276f3
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 54e45faf50d109daaa565b6ae280046d07e7131f
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 54e45faf50d109daaa565b6ae280046d07e7131f
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: f7a4c02f043fa5ad85e3359ae1ef4411243c991f
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: f7a4c02f043fa5ad85e3359ae1ef4411243c991f
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com
